File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;
class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        for (int s : scoville) {
            pq.add(s);
        }
        
        while (true) {
            if (pq.size() == 1) {
                if (pq.peek() < K) return -1;
            }
            
            int one = pq.poll();
            if (one >= K) break;
            int two = pq.poll();
            
            int t = one + two * 2;
            pq.add(t);
            answer++;
        }
        
        
        
        return answer;
    }
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.cpp
#include <bits/stdc++.h>

using namespace std;

priority_queue<int, vector<int>, greater<int>> pq;

int solution(vector<int> scoville, int K) {
    int answer = 0;
    int all_sum = 0;
    
    for (int n : scoville) {
        pq.push(n);
        all_sum += n;
    }
    
    while (pq.top() < K) {
        if (pq.size() == 1) return -1;
        answer++;
        int one = pq.top(); pq.pop();
        int two = pq.top(); pq.pop();
        
        int new_one = one + two * 2;
        pq.push(new_one);
    }
    
    return answer;
}